packages/compass-components/src/components/drawer-portal.tsx

```diff
@@ -321,3 +321,14 @@ export function useDrawerActions() {
   });
   return stableActions.current;
 }
+
+export const useDrawerState = () => {
+  const drawerToolbarContext = useDrawerToolbarContext();
+  const drawerState = useContext(DrawerStateContext);
+  return {
+    isOpen:
+      drawerToolbarContext.isDrawerOpen &&
+      // the second check is a workaround, because LG doesn't set isDrawerOpen to false when it's empty
+      drawerState.length > 0,
+  };
+};
```

Comment:
If we have to add this, we have to do more wiring to actually make it work in our paradigm, this hook will only work if you're using this INSIDE leafygreen layout provider, but not inside our drawer provider, which will not be always true in our case. If you want to add a hook like that you have to first move this state higher up to our provider and re-expose it from there